packages/gradle/src/utils/version-catalog-utils.ts

```diff
@@ -0,0 +1,189 @@
+    }
+  }
+
+  return versionCatalogPaths;
+}
+
+export function updatePluginVersionInCatalog(
+  catalogContent: string,
+  pluginName: string,
+  newVersion: string
+): string {
+  const toml = TOML.parse(catalogContent);
+
+  // Handle plugins section
+  if (toml.plugins) {
+    for (const [pluginAlias, pluginConfig] of Object.entries(toml.plugins)) {
+      if (typeof pluginConfig === 'string') {
+        // Handle simple format: plugin = "id:version"
+        if (pluginConfig.startsWith(`${pluginName}:`)) {
+          toml.plugins[pluginAlias] = `${pluginName}:${newVersion}`;
+          break;
+        }
+      } else if (typeof pluginConfig === 'object' && pluginConfig !== null) {
+        // Handle object format: { id = "plugin.name", version = "1.0.0" }
+        const config = pluginConfig as any;
+        if (config.id === pluginName) {
+          if (config.version && typeof config.version === 'string') {
+            // Direct version
+            config.version = newVersion;
+          } else if (
+            config.version &&
+            typeof config.version === 'object' &&
+            config.version.ref &&
+            toml.versions
+          ) {
+            // Version reference in object form: version = { ref = "nx-version" }
+            const versionRef = config.version.ref;
+            if (toml.versions[versionRef]) {
+              toml.versions[versionRef] = newVersion;
+            }
+          } else if (config['version.ref'] && toml.versions) {
+            // Version reference - update the referenced version
+            let versionRef = config['version.ref'];
+            // Handle the case where version.ref might be an object with ref property
+            if (typeof versionRef === 'object' && versionRef.ref) {
+              versionRef = versionRef.ref;
+            }
+            if (toml.versions[versionRef]) {
+              toml.versions[versionRef] = newVersion;
+            }
+          }
+          break;
+        }
+      }
+    }
+  }
+
+  return TOML.stringify(toml);
+}
+
+export function hasPluginInCatalog(
+  catalogContent: string,
+  pluginName: string
+): boolean {
+  const toml = TOML.parse(catalogContent);
+
+  if (!toml.plugins) {
+    return false;
+  }
+
+  for (const pluginConfig of Object.values(toml.plugins)) {
+    if (typeof pluginConfig === 'string') {
+      // Handle simple format: plugin = "id:version"
+      if (pluginConfig.startsWith(`${pluginName}:`)) {
+        return true;
+      }
+    } else if (typeof pluginConfig === 'object' && pluginConfig !== null) {
+      // Handle object format: { id = "plugin.name", version = "1.0.0" }
+      const config = pluginConfig as any;
+      if (config.id === pluginName) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+export function extractPluginVersionFromCatalog(
+  catalogContent: string,
+  pluginName: string
+): string | null {
+  const toml = TOML.parse(catalogContent);
```

Comment:
Each of these util functions parses the catalog content...

Can the utils take a parsed toml file? so it can be parsed once?